Sources/UI/Chat/Cells/MessageTextEnrichment.swift

```diff
@@ -68,10 +67,15 @@ final class MessageTextEnrichment {
     func enrich() -> Observable<NSAttributedString> {
         Observable.create({ [weak self] observer -> Disposable in
             if let self = self {
-                self.parse()
-                self.parseLinks()
+                if let cachedAttributedString = self.cachedEnrich() {
+                    self.attributedString = NSMutableAttributedString(attributedString: cachedAttributedString)
+                } else {
+                    self.parse()
+                    self.parseLinks()
+                }
```

Comment:
Since we have this logic, why do we need to have `cachedEnrichText` as another function? We don't need to introduce that function, caching will be handled here

Regarding my comment on cache keys/function names, this can even be moved to initializer

Reply:
We want that to get the cache value without triggering the parsing if it's not cached. See `ChatViewController+Cells.swift` line 47. It basically "hey, use the cached value if it's already there, but let's not parse anything here"

Reply:
If the caching logic is handled inside `MessageTextEnrichment` we don't need to have "hey, give me cached if you have it, or parse it" we can just have "hey, give me the value" and `MessageTextEnrichment` can handle it internally

Reply:
@b-onc sorry... got caught up with other tasks last week.

Not sure what you're suggesting. What would be in `ChatViewController+Cells.swift` line 47 if `cachedEnrichText` does not exist? Do we want parsing to happen at cellForRowAt time? The logic there is to have attributedText set if there's a cached value.

Reply:
@cardoso No worries, can you please register this branch to our repo so we can take over? It's hard to finalize this PR with the time difference.

Reply:
Thanks! Closing this one in favor of #243